lib/galaxy/datatypes/tabular.py

```diff
@@ -1362,6 +1362,12 @@ class CMAP(TabularData):
     file_ext = "cmap"
 
     def sniff_prefix(self, file_prefix):
+        start = file_prefix.string_io().read(3000).strip().split('\n')
+        for line in start:
+            if '# CMAP File Version' in line:
+                return True
+            else:
+                pass
         return file_prefix.startswith('# CMAP File Version:')
```

Comment:
What about the following, which should return earlier:

```suggestion
        handle = file_prefix.string_io()
        for line in handle:
            if not line.startswith('#'):
                return False
            if line.startswith('# CMAP File Version:')
                return True
        return False
```

Reply:
That looks like a better idea, yeah. Thanks!